internal/p2p/router_test.go

```diff
@@ -114,7 +117,10 @@ func TestRouter_Channel_Basic(t *testing.T) {
 	)
 	require.NoError(t, err)
 
-	require.NoError(t, router.Start())
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	require.NoError(t, router.Start(ctx))
 	t.Cleanup(func() {
```

Comment:
This was pre-existing, but I'm not sure it's safe to fatal in a cleanup handler—fails after the test case exits usually trigger a (intentional) panic in the test harness I think.

I suspect this really wants to be `defer func() { require.NoError(t, router.Stop()) }()` or similar. (We have probably never seen it fail)